File: Python/listacompras.py
# programa de lista de compras
from numpy import * 
import os

def parada():
  cont = 0
  while True :
    yield cont
    cont += 1
    
def cls():
  os.system("cls")


p = array([])
v = array([])

print('Lista de compras do mês.\n')
mes_str = input('Digite o mes que estamos:  ')

for y in parada() :
  print('Digite o nome do produto:  ')
  produto = input()
  p = append(p, produto)
  print('Digite o preço:   ')
  valor = float(input())
  v = append(v, valor)
  y = input('Continuar?(s/n):  ')
  if y == 'n' :
    break

cls()

print("Tabela de compras do mes ", mes_str, "\n")
print('{:<10}'.format('PRODUTO'), 'VALOR')
for i,j in zip(p, v):
 print('{:<10}'.format(i),'{:.2f}'.format(j))

# LJ

File: C++/listacompras.cpp
#include <iostream>
using std::cin;
using std::cout;
using std::endl;

#include <cstdio>

#include <iomanip>
using std::fixed;
using std::ios;
using std::setprecision;
using std::setw;

int main()
{

  int i, n, s, j;
  char mes[20], produto[30][30];
  float valor_total, total, valor[30];
  char ch;

  cout << "Lista de compras do mês.\n";
  cout << "Digite o mes que estamos:  ";
  cin >> mes;

  for (;;)
  {
    cout << "Digite o nome do produto:  ";
    cin >> produto[i];

    cout << "Digite o preço do produto:  ";
    cin >> valor[i];

    valor_total += valor[i];
    total = static_cast<float>(valor_total);

    fflush(stdin);

    cout << "Continuar? (s//n)  \n";
    cin >> ch;

    if (ch == 'n')
    {
      break;
    }
    i++;
  }
  i++;

  system("clear");

  cout << "Tabela de compras do mês" << mes;
  cout << "PRODUTO      PREÇO\n";

  for (j = 0; j < i; j++)
  {
    cout << produto[j] << setw(15) << setprecision(2) << fixed << valor[j] << endl;
  }
  cout << "\nTOTAL:" << setw(6)
       << valor_total;

  return 0;
}